packages/steps/src/BuildWorkflow.ts

```diff
@@ -19,7 +19,18 @@ export class BuildWorkflow {
     let maybeError: Error | null = null;
     let hasAnyPreviousStepFailed = false;
     for (const step of this.buildSteps) {
-      if (step.shouldExecuteStep(hasAnyPreviousStepFailed)) {
+      let shouldExecuteStep = false;
+      try {
+        shouldExecuteStep = step.shouldExecuteStep(hasAnyPreviousStepFailed);
+      } catch (err: any) {
```

Comment:
The config that the user was using was:
```yml
- eas/send_slack_message:
        if: ${ success() }
        name: Send Slack message on success
        inputs:
          payload:
            blocks:
              - type: header
                text:
                  type: plain_text
                  text: 'Company - :apple_logo: iOS - Develop Pipeline :large_green_circle:'
                  emoji: true
              - type: section
                text:
                  type: mrkdwn
                  text: |-
                    *Build Number:* ${ eas.job.version.buildNumber }
              - type: divider
              - type: section
                text:
                  type: mrkdwn
                  text: |-
                    *Status:* Building and passing all tests.
                    *Install URL:* ${ eas.job.expoBuildUrl }
    - eas/send_slack_message:
        if: ${ failure() }
        name: Send Slack message on failure
        inputs:
          payload:
            blocks:
              - type: header
                text:
                  type: plain_text
                  text: 'Company - :apple_logo: iOS - Develop Pipeline :red_circle:'
                  emoji: true
              - type: section
                text:
                  type: mrkdwn
                  text: |-
                    *Build Number:* ${ eas.job.version.buildNumber }
              - type: divider
              - type: section
                text:
                  type: mrkdwn
                  text: |-
                    *Status:* Build failed, see job for more information
                    *Link:* ${ eas.job.expoBuildUrl }
```

I believe that `step.shouldExecuteStep` is only about evaluating the if condition and it is trivial `success()` or `failure()` in the bug report. 
https://github.com/expo/eas-build/blob/e5d693eb29a484eda5306163df03b0369e31a39d/packages/steps/src/BuildStep.ts#L285-L317

The `try/catch` you added is a good improvement and it seems like it can really save us from erroring in some cases, but I understand it is not related to the bug report itself and it wasn't the root cause of this bug, right? We probably have some other interpolation issues when interpolating the job object.

Reply:
What config did you use to replicate the issue?

Reply:
The user who reported this said
```
however I comment out the ${ eas.job.expoBuildUrl } line as --local builds do not have a build url
```
so I took an existing config for sending slack messages and commented out a line being interpolated with non-existing value - just as the user described. And it resulted in the same problem

Reply:
Unless we want to do some other action when the user tries to interpolate non-existing value then throwing and error

Reply:
`shouldExecuteStep` never returns in the case described by the user, it throws an error